test/sync/src/androidTest/kotlin/io/realm/test/mongodb/shared/SyncSessionTests.kt

```diff
@@ -287,6 +291,89 @@ class SyncSessionTests {
         }
     }
 
+    /**
+     * - Insert a document in MongoDB using the command server
+     * - Fetch the object id of the newly inserted document
+     * - Open a Realm with the same partition key as the inserted document
+     * - Wait for Sync to fetch the document as a valid RealmObject with the matching ObjectId as a PK
+     */
+    @Test
+    fun syncingObjectIdFromMongoDB() {
+        val adminApi = app.asTestApp
+        runBlocking {
+            val config = SyncConfiguration.Builder(user, partitionValue, schema = setOf(ObjectIdPk::class))
+                .build()
+            val realm = Realm.open(config)
+
+            val json: JsonObject = adminApi.insertDocument(
+                ObjectIdPk::class.simpleName!!,
+                """
+                    {
+                        "name": "$partitionValue",
+                        "realm_id" : "$partitionValue"
+                    }
+                """.trimIndent()
+            )
+            val oid = json["insertedId"]?.toString()?.removeSurrounding("\"")
```

Comment:
I believe you can use `json["insertedId"].jsonPrimitive.asString` or something like that instead of this work-around.